libraries/heft-config-file/src/ConfigurationFile.ts

```diff
@@ -395,7 +401,7 @@ export class ConfigurationFile<TConfigurationFile> {
       });
     }
 
-    let parentConfiguration: Partial<TConfigurationFile> = {};
```

Comment:
Hmm this should still be `Partial`, shouldn't it?

Reply:
The way I see it, if we don't have an `extends` field, there is no parent configuration file, so it should be `undefined`. Below we pass in an empty object in the case that it's still `undefined` in order to satisfy the `Partial<TConfigurationFile>` arg, but that's not strictly the "parent configuration file", more an artifact of how we call the method.

Likewise, `_mergeObjects` will only take what it assumes are partial values for the field, and return a partial result, since we can't be sure that the type is entirely complete. The _only_ time that we strictly know that everything is complete is after we've validated against a provided schema, since that's the source of truth for the object structure.